mantidproject/mantid
# Catch errors opening the sliceviewer on MatrixWorkspaces types that don't contain expected information

**Description of work.**

Refactors some code around the image info widget and fixes an issue with retrieving efixed from grouped detectors on indirect instruments.

**To test:**

* Use the file in the related issue and see that the sliceviewer now opens.
* Use other workspace types, e.g from #28832 and other MatrixWorkspace types from other reductions and check they load. Others to check would be results from `Testing/SystemTests/tests/analysis/reference`. 

Fixes #28966 

*This does not require release notes* because **it was a regression introduced this development cycle.**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
